Source/TestDafny/README.md

```diff
@@ -0,0 +1,37 @@
+# TestDafny (a.k.a %testdafny)
+
+This internal, testing-only utility is used to execute a single Dafny source file against each compile target option and assert that the output
+is the same as the corresponding `<test file>.expect` file. This is a very common testing approach implemented manually across the integration test suite with multiple
+lit commands. See [this test file](../../Test/metatests/ConsistentWhenSupported.dfy) for a comparison of the manual approach
+compared to using this utility.
+
+Note the manual approach has several downsides:
+
+1. New compilers have to be manually added to every relevant test file.
+2. The expected output must be duplicated for each compile target, making it hard to perceive inconsistencies.
```

Comment:
I'm surprised about this part: can't we write the following?

```
dafny /compileTarget:cs /compile:3 > %t.cs-output
DIFF %t.cs-output ….expect
dafny /compileTarget:java /compile:3 > %t.java-output
DIFF %t.java-output ….expect
dafny /compileTarget:go /compile:3 > %t.go-output
DIFF %t.go-output ….expect
…
```

Reply:
That's a valid point, and would be a slightly better version of the existing pattern. I'll rephrase slightly.